codeDeploy/elb/common_functions.sh

```diff
@@ -116,26 +298,30 @@ autoscaling_enter_standby() {
     if [ -z "$min_cap" -o -z "$desired_cap" ]; then
         msg "Unable to determine minimum and desired capacity for ASG ${asg_name}."
         msg "Attempting to put this instance into standby regardless."
+        set_flag "asgmindecremented" "false"
     elif [ $min_cap == $desired_cap -a $min_cap -gt 0 ]; then
         local new_min=$(($min_cap - 1))
         msg "Decrementing ASG ${asg_name}'s minimum size to $new_min"
         msg $($AWS_CLI autoscaling update-auto-scaling-group \
-            --auto-scaling-group-name "${asg_name}" \
+            --auto-scaling-group-name \"${asg_name}\" \
             --min-size $new_min)
         if [ $? != 0 ]; then
             msg "Failed to reduce ASG ${asg_name}'s minimum size to $new_min. Cannot put this instance into Standby."
             return 1
         else
-            msg "ASG ${asg_name}'s minimum size has been decremented, creating flag file /tmp/asgmindecremented"
-            # Create a "flag" file to denote that the ASG min has been decremented
-            touch /tmp/asgmindecremented
+            msg "ASG ${asg_name}'s minimum size has been decremented, creating flag in file $FLAGFILE"
+            # Create a "flag" denote that the ASG min has been decremented
+            set_flag "asgmindecremented" "true"
         fi
+    else
+        msg "No need to decrement ASG ${asg_name}'s minimum size"
+        set_flag "asgmindecremented" "false"
     fi
 
     msg "Putting instance $instance_id into Standby"
     $AWS_CLI autoscaling enter-standby \
         --instance-ids $instance_id \
-        --auto-scaling-group-name "${asg_name}" \
+        --auto-scaling-group-name \"${asg_name}\" \
         --should-decrement-desired-capacity
     if [ $? != 0 ]; then
```

Comment:
Here and on Line 334 we should add in the graceful fail...
```
   if [ "$HANDLE_PROCS" = "true" ]; then
        # Resume processes, except for the ones suspended before deployment
        resume_processes
    fi
```
then we're ok if the enter standby function fails.